source/core/types/ut_output_dbms_output.tpb

```diff
@@ -0,0 +1,34 @@
+create or replace type body ut_output_dbms_output as
+
+  constructor function ut_output_dbms_output(self in out nocopy ut_output_dbms_output) return self as result is
+  begin
+    self.output_type := $$plsql_unit;
+    self.output_id := self.generate_output_id();
+    return;
+  end;
+
+  overriding member procedure open(self in out nocopy ut_output_dbms_output) is
+  begin
+    null;
+  end;
+
+  overriding member procedure send(self in out nocopy ut_output_dbms_output, a_text clob) is
+    l_text_part varchar2(32767 byte);
+    --we're assuming max of 2 bytes per char
+    c_size_limit_chars constant integer := (32767/2);
+    i integer := 0;
+  begin
+    while i <= length(a_text) loop
```

Comment:
I think here "less" in neede instead of "less or equal" as if the string length is equal to `c_size_limit_chars` an additional empty line will appear.